core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -228,6 +230,13 @@ public void initialize(String name, Map<String, String> unresolved) {
               client, tokenRefreshExecutor(), token, expiresAtMillis(mergedProps), catalogAuth);
     }
 
+    this.restPageSize = mergedProps.get(REST_PAGE_SIZE);
```

Comment:
Looking at where we ended up here, I think it make smore sense now to use:

```java
import org.apache.iceberg.util.PropertyUtil;

this.restPageSize =PropertyUtil.propertyAsNullableInt(mergedProps, REST_PAGE_SIZE);
```

This avoids needing to do the parsing and lets the property util class expose the parse error.

Reply:
Will fix